### ISSUE TITLE
[BUG] Delta stream not stopping after being suppresed

### ISSUE THREAD

**Issue by RNGezeus** [pagefaultgames/pokerogue#2389]
**Describe the bug**
Weather not being removed upon being suppressed

**To Reproduce**
Give opponent Delta stream and yourself Gastro acid along with a move that would be super effective against them

**Expected behavior**
Strong winds disappearing after the ability is suppresed, thus making the attack super effective and being able to setup new weather.

**Screenshots / Videos**
Ability not cancelling after being suppressed
https://github.com/pagefaultgames/pokerogue/assets/46976469/1230c41c-b2e9-49a5-a474-e21e9140e545

**Additional context**
https://bulbapedia.bulbagarden.net/wiki/Delta_Stream_(Ability)
" If the only Pokémon on the field with Delta Stream is rotated out or switched out, or has its Ability suppressed or replaced, the strong winds will dissipate. "

**Comment by zacharied** [comment]
Yep, it seems like turning off the effect is manually handled by the ability (see [`PreSwitchOutClearWeatherAbAttr`](https://github.com/pagefaultgames/pokerogue/blob/04c94fcbbac568321bddd086c0b038e2fd6e5c8b/src/data/ability.ts#L2067)). So, there's a few options:

* Rename `PreSwitchOutClearWeatherAbAttr` to `EffectEndClearWeatherAbAttr` or some shit like that. We'd probably want to go ahead and rename all the other `PreSwitchOut` effects along with it.
* Add a new `PreAbilitySuppressedAbAttr` that is fired on suppression, and a descendant `PreAbilitySuppressedClearWeatherAbAttr` that implements the weather clear effect.
* Just fire `PreSwitchOutAbAttr` when suppression happens.

Opinions would be appreciated.

**Comment by Snailman11** [comment]
https://github.com/user-attachments/assets/e16d2df1-f788-4118-b36f-c472e25f6aaa

- Neutrallizing gas doesn't suppress the strong winds part either
